packages/x-data-grid-pro/src/components/headerFiltering/GridHeaderFilterCell.tsx

```diff
@@ -324,15 +316,22 @@ const GridHeaderFilterCell = React.forwardRef<HTMLDivElement, GridHeaderFilterCe
 
     const isFilterActive = isApplied || hasFocus;
 
+    let style = {
+      height,
+      width,
+      ...styleProp,
+    };
+
+    const pinnedSide = rtlFlipSide(pinnedPosition, isRtl);
+    if (pinnedSide && pinnedOffset !== undefined) {
+      style = { ...style, [pinnedSide]: pinnedOffset };
+    }
```

Comment:
This block of code owns `style`, we could instead assign a value to it directly rather than making another copy.